src/components/EnvelopeList.vue

```diff
@@ -459,15 +459,15 @@ export default {
 			}
 			this.unselectAll()
 		},
-		markSelectionNotJunk() {
-			this.selectedEnvelopes.forEach((envelope) => {
+		async markSelectionNotJunk() {
+			for (const envelope of this.selectedEnvelopes) {
 				if (envelope.flags.$junk) {
 					this.$store.dispatch('toggleEnvelopeJunk', {
```

Comment:
```suggestion
					await this.$store.dispatch('toggleEnvelopeJunk', {
```

Reply:
Please also adjust for markSelectionJunk. 
await for dispatch has the result, that the requests are sequential. 
It may take a bit longer, but won't overload the backend with too many requests at once.